src/treetensornetworks/abstracttreetensornetwork.jl

```diff
@@ -29,30 +35,14 @@ function set_ortho_region(tn::AbstractTTN, new_region)
   return error("Not implemented")
 end
 
-# 
-# Orthogonalization
-# 
-
-function ITensorMPS.orthogonalize(tn::AbstractTTN, ortho_center; kwargs...)
-  if isone(length(ortho_region(tn))) && ortho_center == only(ortho_region(tn))
-    return tn
-  end
-  # TODO: Rewrite this in a more general way.
-  if isone(length(ortho_region(tn)))
-    edge_list = edge_path(tn, only(ortho_region(tn)), ortho_center)
-  else
-    edge_list = post_order_dfs_edges(tn, ortho_center)
-  end
-  for e in edge_list
-    tn = orthogonalize(tn, e)
+function ITensorMPS.orthogonalize(ttn::AbstractTTN, region::Vector; kwargs...)
+  new_path = post_order_dfs_edges_region(ttn, region)
+  existing_path = post_order_dfs_edges_region(ttn, ortho_region(ttn))
```

Comment:
Change this logic. Create a contiguous super region containing `ortho_region` and at least one vertex `v` in `region`. Do `post_order_dfs_edges(ttn[super_region], v)`.